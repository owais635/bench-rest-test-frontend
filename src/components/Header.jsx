import React from "react";
import styled from "styled-components";

const StyledHeader = styled.header`
  background: ${(props) => props.theme.primary};
  padding: 8px;
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default function Header({ title = "" }) {
  return (
    <StyledHeader>
      <p>{title}</p>
    </StyledHeader>
  );
}
