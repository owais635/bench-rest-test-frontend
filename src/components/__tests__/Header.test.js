import React from "react";
import renderer from "react-test-renderer";
import Header from "../Header";

describe("Header", () => {
  test("should render with a title", () => {
    const component = renderer.create(<Header title="This is a title" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render without a title", () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
