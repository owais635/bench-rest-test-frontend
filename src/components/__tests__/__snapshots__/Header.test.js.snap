// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render with a title 1`] = `
<header
  className="sc-bdvvtL eiMmBc"
>
  <p>
    This is a title
  </p>
</header>
`;

exports[`Header should render without a title 1`] = `
<header
  className="sc-bdvvtL eiMmBc"
>
  <p>
    
  </p>
</header>
`;
